packages/plugin/webpack/src/WebpackPlugin.ts

```diff
@@ -160,61 +167,185 @@ export default class WebpackPlugin extends PluginBase<WebpackPluginConfig> {
-                title: `Building webpack bundle for ${chalk.magenta(pArch)}`,
-                task: async (_, task) => {
-                  return task.newListr(
-                    [
-                      {
-                        title: 'Preparing native dependencies',
-                        task: async (_, innerTask) => {
-                          await listrCompatibleRebuildHook(
-                            this.projectDir,
-                            await getElectronVersion(this.projectDir, await fs.readJson(path.join(this.projectDir, 'package.json'))),
-                            platform,
-                            pArch,
-                            config.rebuildConfig,
-                            innerTask
-                          );
-                        },
-                        options: {
-                          persistentOutput: true,
-                          bottomBar: Infinity,
-                          showTimer: true,
-                        },
-                      },
-                      {
-                        title: 'Building webpack bundles',
-                        task: async () => {
-                          await this.compileMain();
-                          await this.compileRenderers();
-                          // Store it in a place that won't get messed with
-                          // We'll restore the right "arch" in the afterCopy hook further down
-                          const targetDir = path.resolve(this.baseDir, pArch);
-                          await fs.mkdirp(targetDir);
-                          for (const child of await fs.readdir(this.baseDir)) {
-                            if (!arches.includes(child)) {
-                              await fs.move(path.resolve(this.baseDir, child), path.resolve(targetDir, child));
-                            }
-                          }
-                        },
-                        options: {
-                          showTimer: true,
-                        },
-                      },
-                    ],
-                    { concurrent: false }
+          const firstArch = arches[0];
+          const otherArches = arches.slice(1);
+
+          const multiArchTasks: ListrTask<NativeDepsCtx>[] =
+            otherArches.length === 0
+              ? []
+              : [
+                  {
+                    title: 'Mapping native dependencies',
+                    task: async (ctx: NativeDepsCtx) => {
+                      const firstArchDir = path.resolve(this.baseDir, firstArch);
+                      const nodeModulesDir = path.resolve(this.projectDir, 'node_modules');
+                      const mapping: Record<string, string[]> = Object.create(null);
+
+                      const webpackNodeFiles = await glob('**/*.node', {
+                        cwd: firstArchDir,
+                      });
+                      const nodeModulesNodeFiles = await glob('**/*.node', {
```

Comment:
In retrospect, this line probably caused https://github.com/electron/forge/issues/3526.